### ISSUE TITLE
Please replace empty link hrefs

### ISSUE THREAD

**Issue by ndarilek** [RocketChat/Rocket.Chat#3045]
I see the following in your changelog:

```
- Remove empty link hrefs
```

Please revert this commit. Empty link hrefs like the ones in packages/rocketchat-ui-sidenav/side-nav make my life as a screen reader user hell. They don't get keyboard focus, can't be interacted with and are sometimes invisible. I pretty much can't use a link with a missing href, so if you could revert this commit and add `href` to any links missing it, that'd be hugely helpful. I know it's not pretty but it is the difference between accessible and inaccessible.

Thanks.

**Comment by ndarilek** [comment]
To be more specific, my Sandstorm rocket.chat instance just updated. As a blind screen reader user, the following is now completely unusable to me via all of the following methods:

 * Tab to control, press Enter
 * Route mouse pointer to control via screen reader, simulate click
 * Use screen reader to search for clickable items and simulate click that way. None of the below are detected as clickables.

The above are all of the ways I have of interacting with controls, so I'm out of options. Here is what no longer works:

 * My account link
 * Online presence links (away/available, etc.)
 * Administration
 * Logout

Further, any other missing href links, such as the links to list more channels, are now also entirely unusable by me. That is no exaggeration, I now can no longer use rocket.chat as a blind screen reader user.

Thanks for your attention, hope we can get a fix for this quickly as I rely on rocket.chat quite a bit.

**Comment by engelgabriel** [comment]
Hi @ndarilek and we are really sorry about that. We will fix it.
If the HREF the only way around this? He have many other buttons with actions, may we use the button role for this?

**Comment by engelgabriel** [comment]
Ref: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_link_role

**Comment by ndarilek** [comment]
If by role you mean tag, sure, you can make them button elements. Just 
please don't leave them as links with extra hrefs, use divs, etc. 
Buttons come with keyboard interactivity baked in and work great.

Thanks.

**Comment by ndarilek** [comment]
ARIA roles for these elements aren't the best idea. If you use a role 
rather than a native element, you generally have to implement correct 
keyboard interactivity (focus, respond to keypresses, etc.) manually. 
The docs/specs say that you should get keyboard interactivity, but that 
support isn't always present in browsers/screen readers.

Basically, if you want a link, then include a href. If you want 
something that opens a new interface component, use a button. But links 
without hrefs, elements with role attributes that make screen readers 
think they're something else, etc. are very much a last resort.

Thanks.

**Comment by ndarilek** [comment]
#3093 goes a long way toward addressing this, though I don't think I caught everything. I'm using grep incantations that occasionally hit minified files that in turn spew screenfuls of minified JS, so I'm sure I missed some. I also caught a few clickable `<div/>` and `<span/>` elements (please don't do this either.)

Further, I don't know if the various "copy settings" buttons still work. Even so, I think this PR should be merged with help cleaning up any buttons that don't work/need better styling. As a screen reader user, I discovered I can't even register an account on Rocket.chat since the empty hrefs were stripped, so it transforms the app from entirely unusable to highly usable once more.

**Comment by ndarilek** [comment]
I think #3093 closes this, or at least addresses all the ones I've found. Have I missed any?

Also, any chance of an updated Sandstorm release with this fix?

Thanks.

**Comment by engelgabriel** [comment]
OK, I'll publish it to Sandstorm on Monday together with releasing version 0.31.0
